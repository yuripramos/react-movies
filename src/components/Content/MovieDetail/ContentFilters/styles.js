import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";
import { grey70 } from "../../../../styles/settings";

export const Label = styled.label`
  color: ${grey70};
  padding: 0 ${rem(15)};
`;

export const Wrapper = styled.div`
  padding: 0 ${rem(15)};
`;

export const FormItem = styled.div`
  padding: 0 ${rem(15)};
  align-items: center;
  display: flex;
  min-width: 100%;
  margin: ${rem(5)} 0;

  ${media.md(css`
    min-width: auto;
    flex-grow: 1;
    max-width: 52%;
    margin: ${rem(10)} 0 0 0;
  `)};
`;
