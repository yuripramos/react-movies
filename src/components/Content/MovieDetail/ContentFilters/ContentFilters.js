import React, { Fragment } from "react";
import { func, object, arrayOf, string, shape } from "prop-types";

import Filter from "../../../common/Filter";
import Dropdown from "../../../common/Dropdown";
import { Label, Wrapper, FormItem } from "./styles";

function ContentFilters({ authorsList, onFilter, defaultFilter }) {
  const componentLabels = {
    filterButton: "APPLY",
    filterLabel: "FILTER BY"
  };
  const authorsArray =
    authorsList &&
    authorsList.map(c => {
      return { name: c.name, value: c.name };
    });
  authorsList && authorsArray.unshift({ name: "", value: "" });
  return (
    <Wrapper>
      {authorsArray && (
        <Filter
          onFilter={onFilter}
          defaultFilterState={defaultFilter}
          componentLabels={componentLabels}
        >
          {(changeFunc, state) => {
            return (
              <Fragment>
                <FormItem>
                  <Label>TYPE</Label>
                  <Dropdown
                    onChange={changeFunc}
                    name="type"
                    list={[
                      {
                        name: "author",
                        value: "author"
                      },
                      {
                        name: "recent",
                        value: "recent"
                      },
                      {
                        name: "oldest",
                        value: "oldest"
                      }
                    ]}
                    value={state && state.filter["type"]}
                    width={100}
                    noEmptySelection
                  />
                </FormItem>
                {state && state.filter["type"] === "author" && (
                  <FormItem>
                    <Label>Author</Label>
                    <Dropdown
                      onChange={changeFunc}
                      name="range"
                      list={authorsArray}
                      value={state && state.filter["range"]}
                      width={230}
                      noEmptySelection
                    />
                  </FormItem>
                )}
              </Fragment>
            );
          }}
        </Filter>
      )}
    </Wrapper>
  );
}

ContentFilters.propTypes = {
  onFilter: func.isRequired,
  defaultFilter: object,
  authorsList: arrayOf(
    shape({
      name: string,
      id: Number
    })
  )
};

export default ContentFilters;
