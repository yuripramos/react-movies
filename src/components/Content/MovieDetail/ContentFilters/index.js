import ContentFilters from "./ContentFilters";

export default ContentFilters;
