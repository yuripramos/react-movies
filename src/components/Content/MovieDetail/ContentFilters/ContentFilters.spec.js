import React from "react";
import { mount } from "enzyme";
import ContentFilters from "./ContentFilters";

const props = {
  onFilter: jest.fn(),
  defaultFilter: {
    type: "author",
    author: "no name"
  }
};

describe("Dashboard Filters component", () => {
  it("should match snapshot", () => {
    expect(mount(<ContentFilters {...props} />)).toMatchSnapshot();
  });
});
