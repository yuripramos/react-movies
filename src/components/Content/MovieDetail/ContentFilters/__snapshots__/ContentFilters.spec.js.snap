// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Filters component should match snapshot 1`] = `
<ContentFilters
  defaultFilter={
    Object {
      "author": "no name",
      "type": "author",
    }
  }
  onFilter={[MockFunction]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-eNQAEJ",
            "isStatic": true,
            "lastClassName": "DLuXk",
            "rules": Array [
              "
  padding: 0 ",
              "0.9375rem",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-eNQAEJ",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-eNQAEJ DLuXk"
      />
    </StyledComponent>
  </styled.div>
</ContentFilters>
`;
